griptape/memory/structure/run.py

```diff
@@ -1,13 +1,17 @@
-import uuid
+from __future__ import annotations
 
-from attrs import Factory, define, field
+from typing import TYPE_CHECKING, Optional
+
+from attrs import define, field
 
-from griptape.artifacts.base_artifact import BaseArtifact
 from griptape.mixins import SerializableMixin
 
+if TYPE_CHECKING:
+    from griptape.artifacts import BaseArtifact
+
 
-@define
+@define(kw_only=True)
 class Run(SerializableMixin):
-    id: str = field(default=Factory(lambda: uuid.uuid4().hex), kw_only=True, metadata={"serializable": True})
```

Comment:
mostly because of the same issue with cloud Threads, an ID cant be specified when creating a Message, so it would be weirdly inconsistent. i made it optional because the id on Run can be populated on load when the cloud API is called

Reply:
This feels like a pretty similar problem as all the other ids (structure ids, structure run ids, event ids). Would be nice if we picked a consistent approach... I want to wrap my head around it.

I see the need to be able to create ids locally without the cloud, but also a need for the cloud to create ids to ensure uniqueness and such.

Reply:
There are certain things like Artifacts that _require_ ids, so we won't have this luxury to remove for those. I think I'd prefer to separate framework generated ids from server generated ids (maybe stored in `meta`?).

Reply:
im thinking we could just remove `id` from `Run` for now. its not used anywhere and is not super valuable imo. question of `ids` throughout the framework is different

Reply:
But I could imagine a similar situation for Artifact in the future.

Going back to this point:
> mostly because of the same issue with cloud Threads, an ID cant be specified when creating a Message, so it would be weirdly inconsistent.

What is this inconsistent with? Where do we allow setting an `id` on a resource and having that map to a cloud resource?

Reply:
Reposting from offline conversation: I don't think we should couple framework ids to server ids. Server ids should be set in `Run.meta` by the Driver. So while `Run.id` has no purpose today, it also doesn't hurt anything and should be kept around.

Reply:
its back